src/plugins/intel_gna/src/backend/gna_limitations.hpp

```diff
@@ -284,178 +153,157 @@ class AbstractValidator {
+     * @param gna_compile_target GNA compile target
+     * @param gna_precision GNA inference precision
+     */
+    void CheckAllOpsSupported(const std::shared_ptr<ov::Model>& model, const InferenceEngine::Precision gna_precision);
+    bool UseOnly16BitConvolutionWeights() const;
+
+    inline bool IsCropAffinedOffset(size_t numberOfElements) const {
+        const auto cropOffset = numberOfElements * kBytesPerCropElement;
+        return (ALIGN64(cropOffset) != cropOffset);
+    }
+
+    inline size_t GetMemoryAlignment() const {
+        OPENVINO_ASSERT(GetCompileTarget() != target::DeviceVersion::NotSet);
+        return m_mem_alignment;
+    }
+
+    inline std::shared_ptr<cnn2d::AbstractValidator> GetCnnValidator() const {
+        return m_cnn_validator;
+    }
 
-public:
-    Validator_35() = default;
-
-    bool ValidateCnn2D(const std::string& name,
-                       const uint32_t inHeight,
-                       const uint32_t inWidth,
-                       const uint32_t inChannels,
-                       const uint32_t kH,
-                       const uint32_t kW,
-                       const uint32_t kN,
-                       const uint32_t strideH,
-                       const uint32_t strideW,
-                       const uint32_t dilationH,
-                       const uint32_t dilationW,
-                       OvGnaType inPrecision,
-                       bool exception = true) const override;
-
-    bool ValidatePooling2D(const std::string& name,
-                           const uint32_t windowH,
-                           const uint32_t windowW,
-                           const uint32_t strideH,
-                           const uint32_t strideW,
-                           bool exception = true) const override;
-
-    bool ValidateInputPadding(const std::string& name,
-                              const uint32_t pad_h_begin,
-                              const uint32_t pad_h_end,
-                              const uint32_t pad_w_begin,
-                              const uint32_t pad_w_end,
-                              const uint32_t kernel_h,
-                              const uint32_t kernel_w,
-                              const bool throwOnError = true) const override;
-
-    bool ShouldUseOnlyConv2DGnaIface() const override;
-
-    bool ValidateCnn1D(const std::string& name,
-                       const uint32_t inHeight,
-                       const uint32_t inWidth,
-                       const uint32_t inChannels,
-                       const uint32_t kH,
-                       const uint32_t kW,
-                       const uint32_t kN,
-                       const uint32_t strideH,
-                       const uint32_t strideW,
-                       const uint32_t dilationH,
-                       const uint32_t dilationW,
-                       OvGnaType inPrecision,
-                       bool exception = true) const override;
+private:
+    Limitations() {}
```

Comment:
```suggestion
    Limitations() {};
```

Reply:
General rule is not to add a semicolon ';' at the end of method implementations within a class. BTW I will move implementation to cpp.